desmod/simulation.py

```diff
@@ -140,7 +142,20 @@ def __enter__(self):
             os.chdir(self.workspace)
 
     def __exit__(self, *exc):
-        os.chdir(self.prev_dir)
+        os.chdir(self.orig_dir)
+        try:
+            os.chdir(self.config['sim.workspace'])
+            self.sync()
+        finally:
+            os.chdir(self.orig_dir)
+
+    def sync(self):
+        if self.config['sim.workspace.s3_sync']:
+            artifacts = []
+            for root, _, files in os.walk(os.curdir, topdown=False):
+                for filename in files:
+                    artifacts.append(os.path.join(root, filename))
```

Comment:
I originally had this logic in `workspacesync/s3.py` and moved it into _Workspace for that exact reason you asked about.  Having the sync backends using the same artifacts input is nice for consistency, code reuse, and computational cycles.  That said, I'm solving a problem that doesn't exist yet, so I'm happy to move it back to `s3.py` for now if you think that is better.